domain/schema/model/sql/0022-resource.sql

```diff
@@ -7,7 +7,7 @@ CREATE UNIQUE INDEX idx_resource_origin_name
 ON resource_origin_type (name);
 
 INSERT INTO resource_origin_type VALUES
-(0, 'uploaded'),
+(0, 'upload'),
```

Comment:
question: why change this? the resource was `uploaded` by a user, not `upload`.

Reply:
I can answer to that, I did the same in my own PR :)

The fact is that there is an Origin enum written by Simon there: [origin.go](https://github.com/juju/juju/blob/b7a80a1457d89742973e989ab5b67c74be166dd6/internal/charm/resource/origin.go#L18) 

It use the term `upload` instead of `uploaded`, so to allows simple conversion between values in db and enum, having the same term is simpler.

On the discussion between `upload` and `uploaded`, I vouched for  `upload` as a noun, to be consistent with the other term of the enum `store`:

* *this resource originated from a **store** *
* *this resource originated from an **upload** *

Using `uploaded` would imply to use an adjective for store, like stored, but it would  be worse, semantically:

* *this resource originated by being **stored** *
* *this resource originated by being **uploaded** *

Reply:
Yea, as Guillaume said, primarily I wanted to match it with the enum in origin.go so that we could work out which `origin_id` to put in the DB from the `charmresource.Origin` field without having to manually match `upload` with `uploaded` to do the check.

Reply:
okay for consistency, does not sound good verbally.